### DISCUSSION TITLE
How do I use Strapi with a database that already has a schema developed?

### DISCUSSION THREAD

**Discussion by nidnogg** [strapi/strapi#6611] [Legacy Topics]
TL;DR: Does strapi require me to model my database schema inside its admin panel?

Or is there a way I can use a database that I created its tables (but didn't populate it yet), populate it then and generate APIs for use in my front-end?

I understand this may be a general misunderstanding on my part as far as and I may have gotten on this framework for the wrong reasons - but I'd like to understand why so I can avoid making such decisions later on and have some better fundamentals for choosing my tooling.

My goal - I'm creating an eCommerce website for an *extremely* small artisan's shop that cannot afford anything but stuff like free AWS tiers at the moment. The end goal is a website for a few products with credit card payment, a database, and reasonably safe authentication, and I want it to be simple enough so that I can eventually scale it if it grows. I want to understand this project very intimately and really get to know my tooling.

What I tried vs my needs - my stack currently consists of react+next on the front-end, and strapi with postgresql for the back-end. I then took to the PostgreSQL docs (and my undergrad's database notes) and tried my hand at designing a database schema based on my app's use cases. However, I believed strapi would work with a traditional database modeled in a bare bones way. What I see now is that the only option from the docs is modeling it from scratch through its admin panel. I'd like to *really* understand out how I'm supposed to work with something I modeled elsewhere - how it all connects. I'd really appreciate any help. I hope this question is not too broad but I'm hoping it'll clear up some huge misconception I have!
